homeassistant/components/waqi/sensor.py

```diff
@@ -142,7 +142,9 @@ def name(self):
     def native_value(self):
         """Return the state of the device."""
         if self._data is not None:
-            return self._data.get("aqi")
+            value = self._data.get("aqi")
+            if not isinstance(value, str):
+                return value
```

Comment:
Maybe something like this? Make sure it is a number? It it an int or a float?

```suggestion
            if value := self._data.get("aqi"):
               with suppress(TypeError):
                  return float(value)
```

Reply:
would this return None if it's `"-"`?

Reply:
Almost except I did make an error, as it is not a TypeError but an `ValueError`:

![CleanShot 2023-08-03 at 21 01 35](https://github.com/home-assistant/core/assets/195327/4240b0e9-7ab9-43bc-92f5-6cd3304deabf)